### DISCUSSION TITLE
New plugin to use AllTalk in Obsidian

### DISCUSSION THREAD

**Discussion by Mithadon** [erew123/alltalk_tts#433] [General]
Hi all,
I've recently discovered Obsidian and, while the variety of available community plugins is great, I could not find a proper local text-to-speech one. This is my solution using the AllTalk API, inspired by the implementation within SillyTavern.
The plugin offers many options, including chunk count per batch, and allows you to generate audio for highlighted text (with right-click context menu) or entire notes. Streaming is available, as is saving concatenated .wav files, with or without automatic embedding into the markdown notes. 

This plugin is designed to take advantage of the XTTS-v2 engine, especially finetuned models, potentially making it quite easy to convert entire stories or libraries. You could generate parts of a sentence or a character's dialogue, for example, with their own unique settings, until satisfied with the output. Then, keep that one .wav file, saved in a folder with the markdown story, and embedded in the text. At the end, you can easily combine every audio clip using other software.

https://github.com/Mithadon/obsidian-alltalk-tts-plugin

Please check it out. I hope that it will be useful to others. Feedback is greatly appreciated - this is my first plugin.
Mith

**Comment by lanavuleta** [comment]
Love it!

**Comment by erew123** [comment]
@Mithadon Very nice :) Dumb question though.... This Obsidian? https://obsidian.md/

**Reply by Mithadon** [reply]
Yep!

**Reply by erew123** [reply]
@Mithadon Awesome... do you want to me to put it on the 3rd Party links on the Wiki... linking to your Github? https://github.com/erew123/alltalk_tts/wiki#-3rd-party-integrations

**Reply by Mithadon** [reply]
@erew123 Oh yeah, that would be great. Thanks :)

**Reply by erew123** [reply]
@Mithadon Done and Done! https://github.com/erew123/alltalk_tts/wiki and I put a short page about it too https://github.com/erew123/alltalk_tts/wiki/Obsidian-Note-&-Knowledge which I assume is ok, but if you want me to change anything, let me know.

Thanks for developing something to work with AllTalk :)

**Reply by Mithadon** [reply]
@erew123 Amazing. Thank you for all the work you've put into it.
